modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-shipment-test/src/testIntegration/java/com/liferay/headless/commerce/admin/shipment/resource/v1_0/test/ShipmentItemResourceTest.java

```diff
@@ -334,6 +405,102 @@ private CommerceOrderItem _getShippableCommerceOrderItem() {
 			RandomTestUtil.randomInt(0, _commerceOrderItems.size() - 1));
 	}
 
+	private void _testPatchShipmentItemWithoutWarehouseId() throws Exception {
```

Comment:
_testPatchShipmentItemWithoutWarehouseId -> _testPatchShipmentItemWithWarehouseExternalReferenceCode

Reply:
in all these new tests can you also verify that the id returned is the same of the "object" where you take the ERC?

Reply:
I'm doing that but in an implicit way. When I put 0 as ID and ERC of the new WH then the object will be retrieved by ERC, but the service layer (as we discussed, should only use ID since the connection is based on the ID) will fetch and automatically set the ID too (based on the ERC). Thus when the object is returned by the get method, it will already contain the new ID and ERC, as well. Thus the need to also reset the new WH ID in the randomPatchShipment because the object that we compare it to (the one we get from the db) will already have the new id based on above logic and not just the ERC, by that we automatically confirm that both the ID and ERC are the same, so there is no need to explicitly compare the two IDs in the test.